src/Microsoft.Azure.SignalR.Common/Auth/JwtBuilders.cs

```diff
@@ -0,0 +1,436 @@
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Security.Claims;
+using Newtonsoft.Json.Bson;
+using System.Linq.Expressions;
+
+namespace Microsoft.Azure.SignalR
+{
+    /* Modified from https://github.com/Azure/azure-sdk-for-net/blob/2999288f431649dcbf54450b7ab237e45c953978/sdk/webpubsub/Azure.Messaging.WebPubSub/src/JwtBuilder.cs
+     * In order to get rid of package `System.IdentityModel.Tokens.Jwt` when generating the JWT token
+     * 
+     * Comparing with the original version, modifications are as follows:
+     *      1. This implementation supports a variable header {"alg":<alg>, "typ":"JWT", "kid":<kid>} while the original version uses a fixed header {"alg":"HS256","typ":"JWT"}
+     *      2. New method `public void AddClaims(IEnumerable<Claim> claims)`
+     *          (a) Follows the logic of https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b1814f6013262793512be596016d5c75bc5d4fea/src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs#L513
+     *          (b) ATTENTION: ONLY supports `Claim.Value` which is string. The original version can handle different                    `ClaimValueType`
+     */
+
+    internal class JwtBuilder : IDisposable
+    {
+        // Registered claims
+        private static byte[] s_nbf = Encoding.UTF8.GetBytes("nbf");
+        private static byte[] s_exp = Encoding.UTF8.GetBytes("exp");
+        private static byte[] s_iat = Encoding.UTF8.GetBytes("iat");
+        private static byte[] s_aud = Encoding.UTF8.GetBytes("aud");
+        private static byte[] s_sub = Encoding.UTF8.GetBytes("sub");
+        private static byte[] s_iss = Encoding.UTF8.GetBytes("iss");
+        private static byte[] s_jti = Encoding.UTF8.GetBytes("jti");
+
+        public static ReadOnlySpan<byte> Nbf => s_nbf;
+        public static ReadOnlySpan<byte> Exp => s_exp;
+        public static ReadOnlySpan<byte> Iat => s_iat;
+        public static ReadOnlySpan<byte> Aud => s_aud;
+        public static ReadOnlySpan<byte> Sub => s_sub;
+        public static ReadOnlySpan<byte> Iss => s_iss;
+        public static ReadOnlySpan<byte> Jti => s_jti;
+
+        private byte[] headerSha256;
+
+        private Utf8JsonWriter _writer;
+        private MemoryStream _memoryStream;
+        private byte[] _key;
+        private bool _isDisposed;
+
+        private byte[] _jwt;
+        private int _jwtLength;
+        private AccessTokenAlgorithm _algorithm;
+
+        public JwtBuilder(byte[] key, int size = 512, string kid = null, AccessTokenAlgorithm algorithm = AccessTokenAlgorithm.HS256)
+        { // typical JWT is ~300B UTF8
+            _jwt = null;
+            _memoryStream = new MemoryStream(size);
+
+            headerSha256 = GenerateHeaderSha256(kid, algorithm, size);
```

Comment:
since we also support Sha512, the method naming and parameter naming does not match